clang/runtime/dpct-rt/include/kernel.hpp.inc

```diff
@@ -86,5 +111,142 @@ static kernel_function_info get_kernel_function_info(const void *function) {
 }
 // DPCT_LABEL_END
 
+static std::vector<char> read_file_to_vector(const std::string &name) {
+  std::ifstream ifs;
+  ifs.open(name, std::ios::in | std::ios::binary);
+  std::vector<char> content((std::istreambuf_iterator<char>(ifs)),
+                            (std::istreambuf_iterator<char>()));
+  return content;
+}
+
+static void write_vector_to_file(const std::string &name, const std::vector<char> &blob) {
+  auto file = std::fstream(name, std::ios::out | std::ios::binary);
+  file.write(blob.data(), blob.size());
+  file.close();
+}
+
+static uint64_t get_size_from_elf(char const *const blob) {
+  if (blob[0] != 0x7F || blob[1] != 'E' || blob[2] != 'L' || blob[3] != 'F')
```

Comment:
The function can not work on Windows, is that expected?

Reply:
Separate get_size_from_dll function was added for Windows.